homeassistant/components/tplink/config_flow.py

```diff
@@ -101,14 +206,70 @@ async def async_step_user(
             errors=errors,
         )
 
+    async def async_step_user_auth_confirm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Dialog that informs the user that auth is required."""
+        errors = {}
+        host = self.context[CONF_HOST]
+        if user_input:
+            credentials = Credentials(
+                user_input[CONF_USERNAME], user_input[CONF_PASSWORD]
+            )
+            try:
+                device = await self._async_try_discover_connect(
+                    host, credentials, raise_on_progress=False
+                )
+            except AuthenticationException:
+                errors["base"] = "invalid_auth"
+            except SmartDeviceException:
+                errors["base"] = "cannot_connect"
+            else:
+                await set_stored_credentials(
+                    self.hass, user_input[CONF_USERNAME], user_input[CONF_PASSWORD]
+                )
+                async_call_later(
+                    self.hass,
+                    REAUTH_REFLOW_CALL_LATER_SECS,
+                    self._async_reload_requires_auth_entries,
+                )
+                return self._async_create_entry_from_device(device)
+
+        return self.async_show_form(
+            step_id="user_auth_confirm",
+            data_schema=STEP_AUTH_DATA_SCHEMA,
+            errors=errors,
+            description_placeholders={CONF_HOST: host},
+        )
+
     async def async_step_pick_device(
         self, user_input: dict[str, Any] | None = None
     ) -> FlowResult:
         """Handle the step to pick discovered device."""
         if user_input is not None:
             mac = user_input[CONF_DEVICE]
             await self.async_set_unique_id(mac, raise_on_progress=False)
-            return self._async_create_entry_from_device(self._discovered_devices[mac])
+            self._discovered_device = self._discovered_devices[mac]
+            host = self._discovered_device.host
+            connection_params = self._discovered_device.connection_parameters
+            self.context[CONF_HOST] = host
+            credentials = await get_stored_credentials(self.hass)
+            # Cannot use the device in discovered_devices here because we need to set the
+            # http.AsyncClient
+            httpx_asyncclient = create_async_httpx_client(self.hass, verify_ssl=False)
```

Comment:
If we use connection parameters as a container for all relevant connection information, we could just set the client to the discovered instance? :-)

Reply:
Each discovered device needs it's own `httpx_asyncclient` as they can't share cookies.  Also ConnectionParams works for one device but not really for the broadcast discovery.

Reply:
That's now done